deepchecks/nlp/text_data.py

```diff
@@ -107,38 +109,51 @@ def __init__(
             raise DeepchecksValueError('raw_text and tokenized_text cannot both be None')
         elif raw_text is None:
             self._validate_tokenized_text(tokenized_text)
-            raw_text = [' '.join(tokens) for tokens in tokenized_text]
+            self._tokenized_text = list(tokenized_text)
+            self._text = [' '.join(tokens) for tokens in tokenized_text]
         elif tokenized_text is None:
             self._validate_text(raw_text)
+            self._text = list(raw_text)
             if self._task_type == TaskType.TOKEN_CLASSIFICATION:
-                tokenized_text = [text.split() for text in raw_text]
+                self._tokenized_text = [sample.split() for sample in self._text]
+            else:
+                self._tokenized_text = None
         else:
             self._validate_text(raw_text)
             self._validate_tokenized_text(tokenized_text)
+            self._text, self._tokenized_text = list(raw_text), list(tokenized_text)
             if len(raw_text) != len(tokenized_text):
                 raise DeepchecksValueError('raw_text and tokenized_text must have the same length')
 
         if index is None:
-            index = np.arange(len(raw_text))
+            self.index = list(range(len(raw_text)))
         elif len(index) != len(raw_text):
             raise DeepchecksValueError('index must be the same length as raw_text')
+        else:
+            self.index = list(index)
```

Comment:
Nirpicking, but why add unnecessary lines of code? was shorter before with less "else" statements and value assigning.